### PR TITLE
feat: add function to check if a list is empty

### LINKED ISSUE #805: Add function to check if a list is empty
**Is your feature request related to a problem? Please describe.**
Currently, there is no function to check if a list is empty. The only way is to use the `count()` function. It works but it is not very (business) friendly. 

```
count(myList) = 0
```

**Describe the solution you'd like**
Add a new function to check if a list is empty.

```
// function signature
is empty(list: List<Any>): Boolean

// examples
is empty([])     // --> true
is empty([1])   // --> false
```

**Related issues**

* Camunda Autormation Platform 7: 
* Zeebe broker:

### PR DESCRIPTION
**Pull request by skayliu**
## Description

Add a new function to check if a list is empty.

## Related issues

closes #805

### CODE REVIEW

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinListFunctionsTest.scala [hunk 1]
```diff
@@ -403,4 +403,11 @@ class BuiltinListFunctionsTest extends AnyFlatSpec with Matchers with FeelIntegr
     eval(""" string join(["foo", 123, "bar"]) """) should be(ValNull)
   }
 
+  "A list is empty() function" should "return if the list is or not empty" in {
+
+    eval(" is empty([]) ") should be(ValBoolean(true))
+    eval(" is empty([ ]) ") should be(ValBoolean(true))
```
Comment (saig0) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinListFunctionsTest.scala:409]: :wrench: This test case is duplicated. `[]` is the same as `[ ]`. 

We can remove one of them.

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinListFunctionsTest.scala [hunk 2]
```diff
@@ -403,4 +403,11 @@ class BuiltinListFunctionsTest extends AnyFlatSpec with Matchers with FeelIntegr
     eval(""" string join(["foo", 123, "bar"]) """) should be(ValNull)
   }
 
+  "A list is empty() function" should "return if the list is or not empty" in {
```
Comment (saig0) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinListFunctionsTest.scala:406]: :wrench: Improve wording.

```suggestion
  "A list is empty() function" should "return if the list is empty" in {
```

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinListFunctionsTest.scala [hunk 3]
```diff
@@ -403,4 +403,11 @@ class BuiltinListFunctionsTest extends AnyFlatSpec with Matchers with FeelIntegr
...
+
+    eval(" is empty([]) ") should be(ValBoolean(true))
+    eval(" is empty([ ]) ") should be(ValBoolean(true))
+    eval(" is empty([1]) ") should be(ValBoolean(false))
+    eval(" is empty([1,2,3]) ") should be(ValBoolean(false))
```
Comment (saig0) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinListFunctionsTest.scala:411]: :wrench: We should add one more test case with named parameters to ensure that the named parameter doesn't change.

```
eval(" is empty(list: [1]) ") should be(ValBoolean(false))
```

**Review by saig0 — CHANGES_REQUESTED** [review]
@skayliu thank you for your contribution. :tada: 

The changes look good. I have a few minor suggestions. Please have a look. :cookie:

**Comment by skayliu** [comment]
Hey, @saig0, it's done,  do i need to rebase to main?

**Review by saig0 — APPROVED** [review]
@skayliu your PR is ready to merge. :rocket:

**Pull request merged**